docs/usage/usage.md

```diff
@@ -701,3 +701,48 @@ Some key facts about VMSS Flex based clusters:
 - In case the configuration of the VMSS will change (e.g. amount of fault domains in a region change; configured in the CloudProfile) all machines of the worker pool need to be rolled
 - It is not possible to migrate an existing primary AvailabilitySet based Shoot cluster to VMSS Flex based Shoot cluster and vice versa
 - VMSS Flex based clusters are using `Standard` SKU LoadBalancers instead of `Basic` SKU LoadBalancers for AvailabilitySet based Shoot clusters
+
+## `BackupBucketConfig`
+
+`BackupBucketConfig` describes the configuration that needs to be passed over for creation of the backup bucket infrastructure. Configuration like immutability (WORM, i.e. write-once-read-many) that can be set on the bucket are specified here. Objects in the bucket will inherit the immutability duration which is set on the bucket, and they can not be modified or deleted for that duration.
+
+This extension supports creating (and migrating already existing buckets) to use [container-level WORM policies](https://learn.microsoft.com/en-us/azure/storage/blobs/immutable-container-level-worm-policies).
+
+Example:
+
+```yaml
+apiVersion: azure.provider.extensions.gardener.cloud/v1alpha1
+kind: BackupBucketConfig
+immutability:
+  retentionType: "bucket"
+  retentionPeriod: 24h
+  locked: false
+```
```

Comment:
```suggestion
```yaml
apiVersion: azure.provider.extensions.gardener.cloud/v1alpha1
kind: BackupBucketConfig
immutability:
  retentionType: "bucket"
  retentionPeriod: 24h
  lockedPolicy: false
```
```

Can we use `lockedPolicy` instead of just `locked` as to me `lockedPolicy` name makes more sense ?

cc @seshachalam-yv @kon-angelo

Reply:
I don't have a strong opinion here. For a boolean value `locked` is okay. 
It will diverge from the GCP spec but let's decide on the correct name and it can be retrofitted if needed.